crates/cache/src/result/search.rs

```diff
@@ -56,6 +56,12 @@ pub struct CachedSearchResult {
     pub results: Arc<HashMap<TableReference, CachedAggregationResult>>,
 }
 
+impl AsTableRefs for CachedSearchResult {
+    fn as_table_refs(&self) -> Arc<HashSet<TableReference>> {
```

Comment:
Can we return an iterator instead to avoid new data allocation?

```
trait AsTableRefs<'a> {
    type Iter: Iterator<Item = &'a TableReference>;
    fn table_refs(&'a self) -> Self::Iter;
}
``